client/transaction-pool/src/revalidation.rs

```diff
@@ -149,6 +158,7 @@ impl<Api: ChainApi> RevalidationWorker<Api> {
 				} else {
 					for xt in &to_queue {
 						extrinsics.remove(xt);
+						self.members.remove(xt);
```

Comment:
Seems that only this required for a fix it.

The rest is defensive programming and logging :)